script.artistslideshow/resources/common/fileops.py

```diff
@@ -0,0 +1,208 @@
+        return True, log_lines
+
+
+def copyFile( src, dst ):
+    log_lines = []
+    if _exists( src ):
+        try:
+            log_lines.append( 'copying file %s to %s' % (src, dst) )
+            _copy( src, dst )
+        except IOError:
+            log_lines.append( 'unable to copy %s to %s' % (src, dst) )
+            return False, log_lines
+        except Exception as e:
+            log_lines.append( 'unknown error while attempting to copy %s to %s' % (src, dst) )
+            log_lines.append( e )
+            return False, log_lines
+        return True, log_lines
+    else:
+        log_lines.append( '%s does not exist' % src )
+        return False, log_lines
+
+
+def deleteFile( src ):
+    return deleteFolder( src, type='file')
+
+
+def deleteFolder( src, type='folder' ):
+    log_lines = []
+    if _exists( src ):
+        if type == 'folder':
+            #in Mac OSX the .DS_Store file, if present, will block a folder from being deleted, so delete the file
+            try:
+                _delete( os.path.join( src, '.DS_Store' ) )
+            except IOError:
+                log_lines.append( 'unable to delete .DS_Store file' )
+            except Exception as e:
+                log_lines.append( 'unknown error while attempting to delete .DS_Store file' )
+                log_lines.append( e )
+            _action = _rmdir
+        else:
+            _action = _delete
+        try:
+            log_lines.append( 'deleting %s %s' % (type, src) )
+            if isXBMC:
+                if not _action( src ):
+                    raise IOError( 'unable to delete item' )
+            else:
+                _action( src )                
+        except IOError:
+            log_lines.append( 'unable to delete %s' % src )
+            return False, log_lines
+        except Exception as e:
+            log_lines.append( 'unknown error while attempting to delete %s' % src )
+            log_lines.append( e )
+            return False, log_lines
+        return True, log_lines
+    else:
+        log_lines.append( '%s does not exist' % src )
+        return False, log_lines
+
+
+def moveFile( src, dst ):
+    log_lines = []
+    cp_loglines = []
+    dl_loglines = []
+    success = False
+    if _exists( src ):
+        cp_success, cp_loglines = copyFile( src, dst )
+        if cp_success:
+            dl_success, dl_loglines = deleteFile( src )
+            if dl_success:
+                success = True
+    else:
+        log_lines.append( '%s does not exist' % src)
+        success = False
+    return success, log_lines + cp_loglines + dl_loglines
+
+
+def atoi( text ):
+    return int(text) if text.isdigit() else text
+
+
+def naturalKeys( text ):
+    '''
+    alist.sort( key=naturalKeys ) sorts in human order
+    http://nedbatchelder.com/blog/200712/human_sorting.html
+    '''
+    return [ atoi( c ) for c in re.split( r'(\d+)', text ) ]
+
+
+def popenWithTimeout( command, timeout ):
```

Comment:
looks like it isn't even used

Reply:
This is a shorter version of my comment above. The libraries in question are commone across all my projects, so they contain some things that aren't always used in a given project but that I use in more than one.  Assuming I'm allowed to keep them, I can delete any unused one, but then that creates new work for me to maintain different versions of the common libraries across my projects.  As before, I'll do that if I have to.